### ISSUE TITLE
TypeError: LocationChangeListener is not a constructor

### ISSUE THREAD

**Issue by Swatinem** [gorhill/uBlock#800]
Both on uBlock Origin and with uMatrix, I get a ton of these typeerrors in the console:

Note that I use the latest firefox nightly.

```
JavaScript error: chrome://umatrix/content/frameScript.js, line 64: TypeError: LocationChangeListener is not a constructor
JavaScript error: chrome://ublock0/content/frameScript.js, line 77: TypeError: LocationChangeListener is not a constructor
```

**Comment by gorhill** [comment]
I just updated to latest Nightly and I also see this now.

**Comment by gorhill** [comment]
Converting the exported variable from `const` to `var` seems to fix the issue. I am tempted to think it's a Nightly issue however.

**Comment by Betsy25** [comment]
This thread on Mozilla's build forum might be related.
http://forums.mozillazine.org/viewtopic.php?f=23&t=2966543

**Comment by streetwolf** [comment]
That post was mine.  Seems Nightly now enforces let and const compliancy.  It breaks quite a few add-ons.

**Comment by streetwolf** [comment]
I changed the two variables you mentioned above from const to var and it fixed some problems I was having where some of my filters were not being honored.

**Comment by Swatinem** [comment]
> That post was mine. Seems Nightly now enforces let and const compliancy. It breaks quite a few add-ons.

Hm, in that case the addon mechanism itself (that `this.EXPORTED_SYMBOLS = ['contentObserver', 'LocationChangeListener'];`) is broken and probably can’t deal with the temporal dead zone yet.

(for let/const, the bindings simply don’t exist before the declaration, whereas with `var`, the bindings exist but are `undefined`)

**Comment by gorhill** [comment]
Fixed in 097489b4e0bd7fb544848e54fe1f0ebeca979631 (bad commit message because haste).

**Comment by Swatinem** [comment]
Thanks for the quick turnaround. :-) Would you also kindly copy the fix over to uMatrix? It suffers from the same error.
